azurerm/data_source_storage_account_sas.go

```diff
@@ -0,0 +1,390 @@
+
+const (
+	connStringAccountKeyKey  = "AccountKey"
+	connStringAccountNameKey = "AccountName"
+	sasSignedVersion         = "2017-07-29"
+)
+
+// This is an ACCOUNT SAS : https://docs.microsoft.com/en-us/rest/api/storageservices/Constructing-an-Account-SAS
+// not Service SAS
+func dataSourceArmStorageAccountSharedAccessSignature() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmStorageAccountSasRead,
+
+		Schema: map[string]*schema.Schema{
+			"connection_string": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+
+			"https_only": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+
+			"resource_types": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"service": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"container": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"object": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"services": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"blob": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"queue": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"table": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"file": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			// Always in UTC and must be ISO-8601 format
+			"start": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			// Always in UTC and must be ISO-8601 format
+			"expiry": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"permissions": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"read": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"write": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"delete": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"list": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"add": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"create": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"update": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"process": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"sas": {
```

Comment:
🤔 is it worth making this `sas_token` or `storage_access_token`?

Reply:
It's not really a token in the uuid/JWT sense... its actually presented as a URL query string... with inconsistent URL escaping in the various parts. Notice only some of the fields are passed to `url.QueryEscape` to be 100% consistent with how MSFT generates and expects these.

I debated including the `?`, but again, the `?` is consistent with the output from the Azure portal... and how most folks will use this, which is to simply concat it to the primary blob end point.